test/Octonion.jl

```diff
@@ -85,57 +164,235 @@ using Test
 
-    @testset "isreal" begin end
-
-    @testset "iszero" begin end
-
-    @testset "isone" begin end
+    @testset "isreal" begin
+        @test isreal(octo(1))
+        @test !isreal(octo(1, 1, 0, 0, 0, 0, 0, 0))
+        @test !isreal(octo(1, 0, 1, 0, 0, 0, 0, 0))
+        @test !isreal(octo(1, 0, 0, 1, 0, 0, 0, 0))
+        @test !isreal(octo(1, 0, 0, 0, 1, 0, 0, 0))
+        @test !isreal(octo(1, 0, 0, 0, 0, 1, 0, 0))
+        @test !isreal(octo(1, 0, 0, 0, 0, 0, 1, 0))
+        @test !isreal(octo(1, 0, 0, 0, 0, 0, 0, 1))
+    end
 
-    @testset "isfinite" begin end
+    @testset "iszero" begin
+        @test iszero(octo(0))
+        @test !iszero(octo(1))
+        @test !iszero(octo(0, 1, 0, 0, 0, 0, 0, 0))
+        @test !iszero(octo(0, 0, 1, 0, 0, 0, 0, 0))
+        @test !iszero(octo(0, 0, 0, 1, 0, 0, 0, 0))
+        @test !iszero(octo(0, 0, 0, 0, 1, 0, 0, 0))
+        @test !iszero(octo(0, 0, 0, 0, 0, 1, 0, 0))
+        @test !iszero(octo(0, 0, 0, 0, 0, 0, 1, 0))
+        @test !iszero(octo(0, 0, 0, 0, 0, 0, 0, 1))
+        @test !iszero(octo(0, 0, 0, 0, 0, 0, 0, 0, true))
+    end
 
-    @testset "isinf" begin end
+    @testset "isone" begin
+        @test isone(octo(1))
+        @test !isone(octo(-1))
+        @test !isone(octo(0, 1, 0, 0, 0, 0, 0, 0))
+        @test !isone(octo(1, 1, 0, 0, 0, 0, 0, 0))
+        @test !isone(octo(1, 0, 1, 0, 0, 0, 0, 0))
+        @test !isone(octo(1, 0, 0, 1, 0, 0, 0, 0))
+        @test !isone(octo(1, 0, 0, 0, 1, 0, 0, 0))
+        @test !isone(octo(1, 0, 0, 0, 0, 1, 0, 0))
+        @test !isone(octo(1, 0, 0, 0, 0, 0, 1, 0))
+        @test !isone(octo(1, 0, 0, 0, 0, 0, 0, 1))
+    end
 
-    @testset "isnan" begin end
+    @testset "isfinite" begin
+        @test isfinite(octo(1:8...))
+        for inf in (Inf, -Inf)
+            @test !isfinite(octo(inf, 0, 0, 0, 0, 0, 0, 0))
+            @test !isfinite(octo(0, inf, 0, 0, 0, 0, 0, 0))
+            @test !isfinite(octo(0, 0, inf, 0, 0, 0, 0, 0))
+            @test !isfinite(octo(0, 0, 0, inf, 0, 0, 0, 0))
+            @test !isfinite(octo(0, 0, 0, 0, inf, 0, 0, 0))
+            @test !isfinite(octo(0, 0, 0, 0, 0, inf, 0, 0))
+            @test !isfinite(octo(0, 0, 0, 0, 0, 0, inf, 0))
+            @test !isfinite(octo(0, 0, 0, 0, 0, 0, 0, inf))
+            @test isfinite(octo(fill(inf, 8)..., true))
+        end
+    end
 
-    @testset "+" begin end
+    @testset "isinf" begin
+        @test !isinf(octo(1:8...))
+        for inf in (Inf, -Inf)
+            @test isinf(octo(inf, 0, 0, 0, 0, 0, 0, 0))
+            @test isinf(octo(0, inf, 0, 0, 0, 0, 0, 0))
+            @test isinf(octo(0, 0, inf, 0, 0, 0, 0, 0))
+            @test isinf(octo(0, 0, 0, inf, 0, 0, 0, 0))
+            @test isinf(octo(0, 0, 0, 0, inf, 0, 0, 0))
+            @test isinf(octo(0, 0, 0, 0, 0, inf, 0, 0))
+            @test isinf(octo(0, 0, 0, 0, 0, 0, inf, 0))
+            @test isinf(octo(0, 0, 0, 0, 0, 0, 0, inf))
+            @test !isinf(octo(fill(inf, 8)..., true))
```

Comment:
Is this expected behavior?
```julia
julia> a = octo(0, 0, 0, 0, 0, 0, 0, Inf,true)
OctonionF64(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Inf, true)

julia> b = octo(0, 0, 0, 0, 0, 0, 0, 1)
Octonion{Int64}(0, 0, 0, 0, 0, 0, 0, 1, false)

julia> isinf(a), isinf(b)
(false, false)

julia> isinf(a*b)
true
```
I thought this seems confusing, so it seems better to replace `@test` with `@test_broken`.

Reply:
> Is this expected behavior?

It's not expected; if a user sets `norm=true` but the fields contain `NaN` or `Inf`, then this is undefined behavior. But it's consistent with how Quaternions currently behaves. i.e. if `norm=true`, assume it is normalized and don't bother checking remaining fields. This test just checks that `norm` is actually checked for the test, which it should be, so I think it should stay as is. Of course, it will be removed with #75, so we could decide to just preemptively not include the test.

`@test_broken` doesn't apply, as if the test passes with `@test_broken`, that's considered a failing test.

Reply:
>@test_broken doesn't apply, as if the test passes with @test_broken, that's considered a failing test.

I thought `isinf(a)` should be `true`, but I think it's ok to keep this line because this will be fixed in #75.